### PR TITLE
Respect sync mode regardless of input state in mailchimp

### PR DESCRIPTION
**Pull request by sherifnada**

### CODE REVIEW

File: airbyte-integrations/connectors/source-mailchimp/source_mailchimp/source.py [hunk 1]
```diff
@@ -82,11 +88,18 @@ def _client(self, config_container: ConfigContainer):
...
+    def _read_record(
+        self, client: Client, configured_stream: ConfiguredAirbyteStream, state: DefaultDict[str, any]
+    ) -> Generator[AirbyteMessage, None, None]:
         entity_map = {
             "Lists": client.lists,
             "Campaigns": client.campaigns,
         }
 
-        for record in entity_map[stream](state=state):
+        stream_name = configured_stream.stream.name
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-mailchimp/source_mailchimp/source.py:99]: this was the only change really needed

File: airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/StandardSourceTest.java [hunk 2]
```diff
@@ -73,9 +73,9 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public abstract class TestSource {
+public abstract class StandardSourceTest {
```
Comment (sherifnada) [inline airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/StandardSourceTest.java:76]: renaming to remove ambiguity

Comment (cgardens) [reply airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/StandardSourceTest.java:76]: ignore this is it's too annoying, but I think the slightly more semantically clear name is SouceStandardTest. It is the "standard tests" that we run for "sources", not the a "standard" version of "source test" (the point being that "source test" isn't really a primitive in our vocab, while "standard tests" and "source" both are).

**Review by cgardens — APPROVED** [review]
lgtm

**Comment by cgardens** [comment]
should we add a test to the StandardTests to catch this case? Can happen separately. If you think it is worth doing would you mind creating an issue for it and putting it into the next sprint?

**Review by jrhizor — APPROVED** [review]

**Pull request merged**